src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Debugging/DefaultCSharpBreakpointResolver.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Reflection;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Debugging
+{
+    // This type is temporary and will be replaced by an ExternalAccess.Razor type once available.
+
+    [Export(typeof(CSharpBreakpointResolver))]
+    internal class DefaultCSharpBreakpointResolver : CSharpBreakpointResolver
+    {
+        private readonly MethodInfo _tryGetBreakpointSpan;
+
+        public DefaultCSharpBreakpointResolver()
+        {
+            try
+            {
+                var assemblyName = typeof(SyntaxTree).Assembly.GetName();
+                assemblyName.Name = "Microsoft.CodeAnalysis.CSharp.Features";
+                var assembly = Assembly.Load(assemblyName);
+                var type = assembly.GetType("Microsoft.CodeAnalysis.CSharp.EditAndContinue.BreakpointSpans");
+                _tryGetBreakpointSpan = type.GetMethod("TryGetBreakpointSpan");
```

Comment:
We make some assumptions below about this method, such as that it returns a bool and takes 4 arguments, can we build those into the `GetMethod` call here so we fail early and explicitly?

Reply:
Good idea but unfortunately because the last parameter is an "out" parameter it doesn't look like I can do that 😢